enterprise/backend/test/metabase_enterprise/transforms/execute_test.clj

```diff
@@ -64,10 +65,56 @@
                     ;; The transforms-test dataset has exactly 4 Gizmo products (IDs 6, 8, 12, 14)
                     ;; First transform filters for categories starting with "G" (Gadget and Gizmo)
                     ;; Second transform filters for category = "Gizmo", resulting in 4 products
-                    (is (=? {:data {:cols [{:name "count"}]
-                                    :rows [[4]]}}
-                            (qp/process-query check-query)))))))))))))
+                    (is (= [[4]]
+                           (mt/formatted-rows [int] query-result)))
+                    (is (= [{:name "count"}]
+                           (map #(select-keys % [:name])
+                                (mt/cols query-result))))))))))))))
+
+(deftest execute-test-aggregation-query
+  (mt/test-drivers (mt/normal-driver-select {:+features [:transforms/table :left-join]})
+    (mt/dataset transforms-dataset/transforms-test
+      (let [target-type "table"
+            schema      (t2/select-one-fn :schema :model/Table (mt/id :transforms_products))]
+        (with-transform-cleanup! [target-table {:type   target-type
+                                                :schema schema
+                                                :name   "widgets_daily"}]
+          (let [mp (mt/metadata-provider)
+                transforms-products (lib.metadata/table mp (mt/id :transforms_products))
+                transforms-orders (lib.metadata/table mp (mt/id :transforms_orders))
+                products-id (lib.metadata/field mp (mt/id :transforms_products :id))
+                orders-product-id (lib.metadata/field mp (mt/id :transforms_orders :product_id))
+                products-category (lib.metadata/field mp (mt/id :transforms_products :category))
+                orders-total (lib.metadata/field mp (mt/id :transforms_orders :total))
+                orders-order-date (lib.metadata/field mp (mt/id :transforms_orders :order_date))
+                aggregated-query (-> (lib/query mp transforms-products)
+                                     (lib/join (lib/join-clause transforms-orders
+                                                                [(lib/= products-id orders-product-id)]))
+                                     (lib/filter (lib/= products-category "Widget"))
+                                     (lib/filter (lib/not-null orders-total))
+                                     (lib/aggregate (lib/sum orders-total))
+                                     (lib/breakout (lib/with-temporal-bucket
+                                                     orders-order-date :day))
+                                     (as-> <> (lib/order-by <> (lib/aggregation-ref <> 0) :desc)))]
+            (mt/with-temp [:model/Transform transform {:name   "transform"
+                                                       :source {:type  :query
+                                                                :query (lib.convert/->legacy-MBQL aggregated-query)}
```

Comment:
```suggestion
                                                                :query aggregated-query}
```

For new models like Transform please persist MBQL 5 in the app DB instead of legacy. It's one less thing we'll have to migrate over in the near future. Everything already works with MBQL 5